### ISSUE TITLE
Display different menu item based on user's role

### ISSUE THREAD

**Issue by orchardc** [OrchardCMS/OrchardCore#3741]
Is it possible to display different menuitem based on user's role?

For example, on the navlink, there are 2 menu items: [Home] [Contact].
But for login user with the role ("Manager"), there should be an additional menu item call [Management]

Something can be added here?

 "MenuItemsListPart": {
            "MenuItems": [
              {
                "ContentType": "LinkMenuItem",
                "ContentItemId": "[js:uuid()]",
                "LinkMenuItemPart": {
                  "Name": "Home",
                  "Url": "~/"
                }
              },
              {
                "ContentType": "LinkMenuItem",
                "ContentItemId": "[js:uuid()]",
                "LinkMenuItemPart": {
                  "Name": "Contact",
                  "Url": "~/contact"
                }
              }
            ]
          },

**Comment by agriffard** [comment]
Would this require a new setting on a MenuItem or a new kind of MenuItem?

**Comment by agriffard** [comment]
Related to #5530

**Comment by vip924** [comment]
Is there any way to show and hide menu items based on the roles given to the content items?

I have created 2 content types and linked them to 2 menu items in the Orchard Core menu content type.

I added a new role, set permission to only a specific content item and removed the view permission of the other content item.

Still, both the content items are shown in the menu. But the 2nd content item is now showing "You do not have access to this resource." in the view page when I click the menu.
